packages/playwright-core/src/client/network.ts

```diff
@@ -172,13 +173,25 @@ export class Request extends ChannelOwner<channels.RequestChannel> implements ap
   }
 
   frame(): Frame {
+    if (!this._initializer.frame) {
+      assert(this.isServiceWorkerRequest());
+      throw new Error('Service Worker requests do not have an associated frame.');
```

Comment:
Frame had been nullable, but then we made it non-nullable at some point in the past. We can't go back, and SW's don't have a frame associated with them.